src/escd/network/peerclientmanager.cpp

```diff
@@ -266,92 +276,82 @@ void PeerConnectManager::connect(std::string peer_address, unsigned short port,
 }
 
 
-void PeerConnectManager::connectPeersFromConfig(int& connNeeded)
+void PeerConnectManager::getPeersFromConfig(std::vector<std::pair<in_addr_t, unsigned short>> &peerAddrs)
 {    
-    for(std::string addr : m_opts.peer)
+    for(std::string address : m_opts.peer)
     {
-        if(connNeeded <= 0){
-            break;
-        }
-
         std::string  peer_address;
         std::string  port;
         std::string  svid;
+        in_addr   addr;
+
+        m_opts.get_address(address, peer_address, port, svid);
 
-        m_opts.get_address(addr, peer_address, port, svid);
+        if(inet_aton(peer_address.c_str(), &addr) != 0 ){
+            peerAddrs.push_back(std::make_pair(addr.s_addr, atol(port.c_str())));
+        } else {
+            boost::asio::ip::tcp::resolver              resolver(m_ioService);
+            boost::asio::ip::tcp::resolver::query       query(peer_address.c_str(),SERVER_PORT);
+            boost::asio::ip::tcp::resolver::iterator    iterator = resolver.resolve(query);
+            boost::asio::ip::tcp::resolver::iterator    end;
 
-        connect(peer_address, atoi(port.c_str()), atoi(svid.c_str()));
-        --connNeeded;
+
+            if(iterator != end)
+            {
+               if(inet_aton(iterator->endpoint().address().to_string().c_str(), &addr) != 0)
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Merge this if statement with the enclosing one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=cpp%3AS1066)